fsw/src/sample_app_cmds.c

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+**
+**      GSC-18128-1, "Core Flight Executive Version 6.7"
+**
+**      Copyright (c) 2006-2019 United States Government as represented by
+**      the Administrator of the National Aeronautics and Space Administration.
+**      All Rights Reserved.
+**
+**      Licensed under the Apache License, Version 2.0 (the "License");
+**      you may not use this file except in compliance with the License.
+**      You may obtain a copy of the License at
+**
+**        http://www.apache.org/licenses/LICENSE-2.0
+**
+**      Unless required by applicable law or agreed to in writing, software
+**      distributed under the License is distributed on an "AS IS" BASIS,
+**      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+**      See the License for the specific language governing permissions and
+**      limitations under the License.
+**
+** File: sample_app_cmds.c
+**
+** Purpose:
+**   This file contains the source code for the Sample App command handling
+**   functions.
+**
+*******************************************************************************/
+
+/*
+** Include Files:
+*/
+#include "sample_app_events.h"
+#include "sample_app_version.h"
+#include "sample_app.h"
+#include "sample_table.h"
+#include "sample_app_cmds.h"
+#include "sample_app_utils.h"
+
+/* The sample_lib module provides the SAMPLE_Function() prototype */
+#include <string.h>
+#include "sample_lib.h"
+
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+/*                                                                            */
+/* SAMPLE_Noop -- SAMPLE NOOP commands                                        */
+/*                                                                            */
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
+void SAMPLE_Noop( const SAMPLE_Noop_t *Msg )
```

Comment:
All command handlers should return an `int32` code, and return `CFE_SUCCESS` if the message was consumed successfully.  Although this is not really used here, it is the recommended pattern because it enables a future "generic" message dispatch routine to take some action (like generating an event or incrementing an error counter) if it fails.

Reply:
The existing apps are pretty split on this.  I have no objection to changing this back.

Reply:
Just like the other patterns, the intent is to allow gradual evolution toward a CFE build solution with generated routing/dispatch tables from some sort of command/data dictionary facility.  So preferable to just return an int32 of `CFE_SUCCESS`.  Even if the dispatcher ignores it today, it could be used in the future.